app/views/welcome/index.html.erb

```diff
@@ -1,4 +1,13 @@
 <h2>Welcome</h2>
 <p>
   The time is now: <%= Time.now %>
-</p>
\ No newline at end of file
+</p>
+
+<%= link_to 'View Topics', topics_url %>
+</br>
+<%= link_to 'Sign Up', new_user_registration_url %>
+</br>
+<%= link_to 'Log In', new_user_session_url %>
+</br>
+<%= link_to 'Log Out', destroy_user_session_url, method: :delete %>
+
```

Comment:
Instead of a list of links that show no matter what, why not conditionally show the appropriate ones. In fact, devise has already done this for you. Check out the login and signup forms and you'll see this line:
```
<%= render "devise/shared/links" %>
```
That will do the trick surely?

Reply:
HA... yeah, that would've been easier :D ... although, annoyingly 'log out' isn't included in the shared links... I've added it in there - this is probably wrong, but devise has them in other directory 'devise/menu/_login_items.html.erb' which I dont have... I've added it to my shared/links file instead, I think my showing of links needs some work. Seeing as I've got everything working and my tests passing I'm going to leave this for another branch.

Reply:
Usually the 'Log out button appears in the menu, so I think that's why the devise one doesn't appear there.